servicex_analysis_utils/cli.py

```
@@ -0,0 +1,44 @@
+import argparse
+import sys
+import json
+import os
+from .file_peeking import get_structure 
+
+def run_from_command():
+    parser = argparse.ArgumentParser(description="CLI tool for retrieving ROOT file structures.")
+
+    parser.add_argument("dataset", nargs='+', help="Input datasets (Rucio DID) or a JSON file containing datasets in a dict.")  
+    parser.add_argument("--filter-branch", default="", help="Only display branches containing this string.")
+    parser.add_argument("--save-to-txt", action="store_true", help="Save output to a text file instead of printing.")
+
+    args = parser.parse_args()
+
+    if len(args.dataset) == 1 and args.dataset[0].endswith(".json"):
+        dataset_file = args.dataset[0]
+        
+        if not os.path.isfile(dataset_file):
+            print(f"\033[91mError: JSON file '{dataset_file}' not found.\033[0m", file=sys.stderr)
+            sys.exit(1)
+
+        try:
+            with open(dataset_file, "r") as f:
+                dataset = json.load(f) 
+                
+                if not isinstance(dataset, dict):
+                    print(f"\033[91mError: The JSON file must contain a dictionary.\033[0m", file=sys.stderr)
+                    sys.exit(1)
+            
+        except json.JSONDecodeError:
```

Don't swallow this error - make sure it gets printed out - it will contains the "reason" the json is bad, which will help the user figure out what they have to fix in their file.

---

Solved with [commit](https://github.com/ssl-hep/ServiceX_analysis_utils/pull/4/commits/01c49a2ec3e17c5a98b023d87caceb2ca019ebb9)

---

Now I see it!